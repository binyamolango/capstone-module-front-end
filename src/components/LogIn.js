import { Button, CircularProgress, TextField } from '@mui/material';
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { createUser } from '../redux/users/usersSlice';
import img from '../images/bg.jpg';

const LogIn = () => {
  const navigate = useNavigate();
  const [name, setName] = useState('');
  const dispatch = useDispatch();
  const [loading, setLoading] = useState(false);

  const handleSubmit = async (e) => {
    e.preventDefault();
    setLoading(true);

    dispatch(createUser({ name }));
    localStorage.setItem('user', JSON.stringify(name));
    setLoading(false);
    navigate('/item');
  };

  const handleUsernameChange = (e) => {
    setName(e.target.value);
  };

  const handleLogIn = () => {
    navigate('/');
  };

  const handleSignUp = () => {
    navigate('/signup');
  };
  return (
    <>
      <div className="flex items-center justify-center h-screen bg-cover bg-center" style={{ backgroundImage: `url(${img})` }}>
        <div className="flex items-center justify-center flex-col gap-6 w-4/5 p-12 md:max-w-fit md:max-h-fit bg-white rounded-md">
          <h1 className="font-bold text-3xl text-center">
            Welcome to
            <span className="text-[#4e8de0] "> Edenic Health</span>
          </h1>
          <h1 className="font-bold text-3xl text-center">
            Log in
          </h1>
          <form className="flex items-center justify-center flex-col gap-6" onSubmit={(e) => handleSubmit(e)}>
            <TextField
              required
              id="outlined-basic"
              type="text"
              onChange={(e) => handleUsernameChange(e)}
              label="Username"
              variant="outlined"
            />
            <div className="flex gap-4">
              <Button type="submit" variant="outlined" disabled={loading} onClick={handleLogIn}>
                {loading ? <CircularProgress size={24} /> : 'Log in'}
              </Button>
              <Button type="submit" variant="outlined" disabled={loading} onClick={handleSignUp}>
                {loading ? <CircularProgress size={24} /> : 'Sign up'}
              </Button>
            </div>
          </form>
        </div>
      </div>
    </>
  );
};

export default LogIn;
