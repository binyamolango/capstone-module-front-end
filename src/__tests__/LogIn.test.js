import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import LogIn from '../components/LogIn';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

let num = 0;
const handleClick = (n) => {
  num = n;
};
describe('Tests LogIn component', () => {
  it('should render page', () => {
    const LogInComponent = renderer.create(
      <BrowserRouter>
        <LogIn />
      </BrowserRouter>,
    );
    expect(LogInComponent).toMatchSnapshot();
  });

  it('should display "Welcome to Edenic Health"', () => {
    render(
      <BrowserRouter>
        <LogIn />
      </BrowserRouter>,
    );

    const h1 = screen.getByText('Welcome to');
    const h2 = screen.getByText('Edenic Health');
    expect(h1).toBeInTheDocument();
    expect(h2).toBeInTheDocument();
  });

  it('should allow click on LogIn button', () => {
    render(
      <BrowserRouter>
        <LogIn />
      </BrowserRouter>,
    );

    const LogInButton = screen.getByRole('button', { name: 'Log in' });
    LogInButton.onclick = handleClick(2);
    fireEvent.click(LogInButton);
    expect(num).toBe(2);
  });

  it('should allow click on signup button', () => {
    render(
      <BrowserRouter>
        <LogIn />
      </BrowserRouter>,
    );

    const signUpButton = screen.getByText('Sign up');
    signUpButton.onclick = handleClick(1);
    fireEvent.click(signUpButton);
    expect(num).toBe(1);
  });

  it('should get value from username field', () => {
    render(
      <BrowserRouter>
        <LogIn />
      </BrowserRouter>,
    );

    const usernameInput = screen.getByRole('textbox');
    expect(usernameInput).toBeInTheDocument();
    fireEvent.change(usernameInput, { target: { value: 'testuser' } });
    expect(usernameInput.value).toBe('testuser');
  });
});
