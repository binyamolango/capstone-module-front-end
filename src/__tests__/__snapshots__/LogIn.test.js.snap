// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests LogIn component should render page 1`] = `
<div
  className="flex items-center justify-center h-screen bg-cover bg-center"
  style={
    Object {
      "backgroundImage": "url(bg.jpg)",
    }
  }
>
  <div
    className="flex items-center justify-center flex-col gap-6 w-4/5 p-12 md:max-w-fit md:max-h-fit bg-white rounded-md"
  >
    <h1
      className="font-bold text-3xl text-center"
    >
      Welcome to
      <span
        className="text-[#4e8de0] "
      >
         Edenic Health
      </span>
    </h1>
    <h1
      className="font-bold text-3xl text-center"
    >
      Log in
    </h1>
    <form
      className="flex items-center justify-center flex-col gap-6"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="outlined-basic"
          id="outlined-basic-label"
        >
          Username
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="outlined-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Username
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="flex gap-4"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Log in
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Sign up
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Tests login component should render page 1`] = `
<div
  className="flex items-center justify-center h-screen bg-cover bg-center"
  style={
    Object {
      "backgroundImage": "url(bg.jpg)",
    }
  }
>
  <div
    className="flex items-center justify-center flex-col gap-6 w-4/5 p-12 md:max-w-fit md:max-h-fit bg-white rounded-md"
  >
    <h1
      className="font-bold text-3xl text-center"
    >
      Welcome to
      <span
        className="text-[#4e8de0] "
      >
         Edenic Health
      </span>
    </h1>
    <form
      className="flex items-center justify-center flex-col gap-6"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="outlined-basic"
          id="outlined-basic-label"
        >
          Username
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="outlined-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Username
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="flex gap-4"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Log in
        </button>
      </div>
    </form>
  </div>
</div>
`;
